drivers/display/display_otm8009a.c

```diff
@@ -0,0 +1,941 @@
+	struct mipi_dsi_device mdev;
+	int ret;
+
+	if (cfg->reset.port) {
+		ret = gpio_pin_configure_dt(&cfg->reset, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Reset display failed! (%d)", ret);
+			return ret;
+		}
+		k_msleep(20);
+		ret = gpio_pin_set_dt(&cfg->reset, 1);
+		if (ret < 0) {
+			LOG_ERR("Enable display failed! (%d)", ret);
+			return ret;
+		}
+		k_msleep(10);
+	}
+
+	/* store x/y resolution & rotation */
+	if (cfg->rotation == 0U) {
+		data->xres = cfg->width;
+		data->yres = cfg->height;
+		data->orientation = DISPLAY_ORIENTATION_NORMAL;
+	} else if (cfg->rotation == 90U) {
+		data->xres = cfg->height;
+		data->yres = cfg->width;
+		data->orientation = DISPLAY_ORIENTATION_ROTATED_90;
+	} else if (cfg->rotation == 180U) {
+		data->xres = cfg->width;
+		data->yres = cfg->height;
+		data->orientation = DISPLAY_ORIENTATION_ROTATED_180;
+	} else if (cfg->rotation == 270U) {
+		data->xres = cfg->height;
+		data->yres = cfg->width;
+		data->orientation = DISPLAY_ORIENTATION_ROTATED_270;
+	}
+
+	/* attach to MIPI-DSI host */
+	mdev.data_lanes = cfg->data_lanes;
+	mdev.pixfmt = data->dsi_pixel_format;
+	mdev.mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST | MIPI_DSI_MODE_LPM;
+
+	mdev.timings.hactive = data->xres;
+	mdev.timings.hbp = OTM8009A_HBP;
+	mdev.timings.hfp = OTM8009A_HFP;
+	mdev.timings.hsync = OTM8009A_HSYNC;
+	mdev.timings.vactive = data->yres;
+	mdev.timings.vbp = OTM8009A_VBP;
+	mdev.timings.vfp = OTM8009A_VFP;
+	mdev.timings.vsync = OTM8009A_VSYNC;
+
+	ret = mipi_dsi_attach(cfg->mipi_dsi, cfg->channel, &mdev);
+	if (ret < 0) {
+		LOG_ERR("MIPI-DSI attach failed! (%d)", ret);
+		return ret;
+	}
+
+	ret = otm8009a_check_id(dev);
+	if (ret) {
+		LOG_ERR("Panel ID check failed! (%d)", ret);
+		return ret;
+	}
+
+	ret = otm8009a_configure(dev);
+	if (ret) {
+		LOG_ERR("DSI init sequence failed! (%d)", ret);
+		return ret;
+	}
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_configure_dt(&cfg->backlight, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not configure bl GPIO (%d)", ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+#define OTM8009A_DEVICE(id)									\
+	static const struct otm8009a_config otm8009a_config_##id = {				\
+		.mipi_dsi = DEVICE_DT_GET(DT_INST_BUS(id)),					\
+		.reset = GPIO_DT_SPEC_INST_GET_OR(id, reset_gpios, {0}),			\
+		.backlight = GPIO_DT_SPEC_INST_GET_OR(id, bl_gpios, {0}),			\
+		.data_lanes = DT_INST_PROP_BY_IDX(id, data_lanes, 0),				\
+		.width = DT_INST_PROP(id, width),						\
+		.height = DT_INST_PROP(id, height),						\
+		.channel = DT_INST_REG_ADDR(id),						\
+		.rotation = DT_INST_PROP(id, rotation),						\
+	};											\
+	static struct otm8009a_data otm8009a_data_##id = {					\
+		.dsi_pixel_format = DT_INST_PROP(id, pixel_format),				\
+	};											\
+	DEVICE_DT_INST_DEFINE(id, &otm8009a_init, NULL, &otm8009a_data_##id,			\
+			      &otm8009a_config_##id, POST_KERNEL,				\
+			      CONFIG_APPLICATION_INIT_PRIORITY, &otm8009a_api);			\
```

Comment:
```suggestion
			      CONFIG_DISPLAY_INIT_PRIORITY, &otm8009a_api);			\
```

Reply:
The problem with this is in my case that the LTDC must be running already. With the same priority this is not guaranteed. I will add a custom INIT_PRIORITY.